jack-core/src/com/rapleaf/jack/transaction/TransactorMetricElement.java

```diff
@@ -0,0 +1,68 @@
+package com.rapleaf.jack.transaction;
+
+
+import java.util.Comparator;
+
+public class TransactorMetricElement {
+
+  private StackTraceElement queryTrace;
+  private long totalExecutionTime;
+  private int count;
+
+  public TransactorMetricElement(StackTraceElement queryTrace, long totalExecutionTime) {
+    this.queryTrace = queryTrace;
+    this.totalExecutionTime = totalExecutionTime;
+    this.count = 1;
+  }
+
+  @Override
+  public int hashCode() {
+    int hash = queryTrace.hashCode();
+    hash += 137 * totalExecutionTime;
+    hash += 197 * count;
+    return hash;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (this == obj) {
+      return true;
+    }
+    if (!(this.getClass() == obj.getClass())) {
```

Comment:
"obj" could be null, and this check could throw null pointer exception. Should use "instanceof", which will return false if obj is null.